src/main/java/com/limechain/trie/BlockTrieAccessor.java

```diff
@@ -14,14 +15,14 @@ public class BlockTrieAccessor extends TrieAccessor {
      *
      * @param blockHash the block hash of the block whose trie is to be accessed
      */
-    public BlockTrieAccessor(Hash256 blockHash) {
-        super(BlockState.getInstance().isInitialized() ?
+    public BlockTrieAccessor(TrieStorage trieStorage, Hash256 blockHash) {
```

Comment:
The `BlockTrieAccessor` is now entirely obsolete IMO, since the problem it was created to solve (circumnavigate sprawling singletons) is solved in this PR. However, this class falls under general between-blocks logic, it's out of scope of the current PR.

Reply:
So it has to be fully deleted?

Reply:
I've absolutely confused the two classes here, I meant to comment that for the `AccessorHolder`. Ignore this discussion.